src/main/java/com/dougnoel/sentinel/steps/TextVerificationSteps.java

```diff
@@ -100,6 +100,36 @@ public static void verifyElementTextContains(String elementName, String assertio
         }
     }
     
+    /**
+     * Waits until we can verify that an element contains certain text. It uses the text
+     * contained in double quotes for matching. If the condition is not true, it will wait 
+     * until the timeout for it to become true, such as if the text on an element changes.
+     * <b>Gherkin Examples:</b>
+     * <ul>
+     * <li>I wait until the header div contains the text "Header"</li>
+     * </ul>
+     * 
+     * @param elementName String The name of the element to be evaluated as defined in the page object.
+     * @param text String The text to verify exists in the element.
+     */
+    @Then("^I wait until the (.*?) contains the text \"([^\"]*)\"$")
+    public static void waitUntilElementTextContains(String elementName, String text) {
+    	Boolean found = getElement(elementName).waitForText(text);	
```

Comment:
Java has primitive and OO types for all base types. The OO types start with a capital letter and allow nulls. In this case this requires you to check for nulls on line 127. Changing this to `boolean` solves the problem. (This is the root cause of the code smell Sonar detected.)
```suggestion
    	boolean found = getElement(elementName).waitForText(text);	
```